pkg/odo/cli/project/project.go

```diff
@@ -2,202 +2,46 @@ package project
 
 import (
 	"fmt"
-	"os"
-	"strings"
 
-	"github.com/redhat-developer/odo/pkg/log"
 	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
 	odoutil "github.com/redhat-developer/odo/pkg/odo/util"
 	"github.com/redhat-developer/odo/pkg/odo/util/completion"
 
-	"github.com/redhat-developer/odo/pkg/project"
 	"github.com/spf13/cobra"
 )
 
-var (
-	projectShortFlag       bool
-	projectForceDeleteFlag bool
-)
-
-var projectCmd = &cobra.Command{
-	Use:   "project [options]",
-	Short: "Perform project operations",
-	Long:  "Perform project operations",
-	Example: fmt.Sprintf("%s\n%s\n%s\n%s\n%s",
-		projectSetCmd.Example,
-		projectCreateCmd.Example,
-		projectListCmd.Example,
-		projectDeleteCmd.Example,
-		projectGetCmd.Example),
-	// 'odo project' is the same as 'odo project get'
-	// 'odo project <project_name>' is the same as 'odo project set <project_name>'
-	Run: func(cmd *cobra.Command, args []string) {
-		if len(args) > 0 && args[0] != "get" && args[0] != "set" {
-			projectSetCmd.Run(cmd, args)
-		} else {
-			projectGetCmd.Run(cmd, args)
-		}
-	},
-}
-
-var projectSetCmd = &cobra.Command{
-	Use:   "set",
-	Short: "Set the current active project",
-	Long:  "Set the current active project",
-	Example: `  # Set the current active project
-  odo project set myproject
-	`,
-	Args: cobra.ExactArgs(1),
-	Run: func(cmd *cobra.Command, args []string) {
-		projectName := args[0]
-		context := genericclioptions.NewContext(cmd)
-		client := context.Client
-		current := context.Project
+// RecommendedCommandName is the recommended service command name
+const RecommendedCommandName = "project"
 
-		exists, err := project.Exists(client, projectName)
-		odoutil.LogErrorAndExit(err, "")
-		if !exists {
-			log.Errorf("The project %s does not exist", projectName)
-			os.Exit(1)
-		}
-
-		err = project.SetCurrent(client, projectName)
-		odoutil.LogErrorAndExit(err, "")
-		if projectShortFlag {
-			fmt.Print(projectName)
-		} else {
-			if current == projectName {
-				log.Infof("Already on project : %v", projectName)
+// NewCmdProject implements the project odo command
+func NewCmdProject(name, fullName string) *cobra.Command {
+
+	projectCreateCmd := NewCmdProjectCreate(createRecommendedCommandName, fullName+" "+createRecommendedCommandName)
```

Comment:
Use `util.GetFullName` here and below instead of concatenating things manually.

Reply:
ack, resolved